crates/oxc_linter/src/rules/no_mixed_operators.rs

```diff
@@ -0,0 +1,344 @@
+        match (node, parent) {
+            (AstKind::BinaryExpression(node), AstKind::BinaryExpression(parent)) => {
+                node.operator != parent.operator
+            }
+            (AstKind::LogicalExpression(node), AstKind::LogicalExpression(parent)) => {
+                node.operator != parent.operator
+            }
+            _ => true,
+        }
+        // Note that there is not need to check for parenthesis explicitly because if an
+        // expression is parenthesized, its parent node is a ParenthesizedExpression and will
+        // never enter the code path.
+    }
+
+    /// Report mixed operator pare between node and parent corresponding to configuration.
+    fn report(&self, node: AstKind, parent: AstKind, ctx: &LintContext<'_>) {
+        let (node_operator, node_left_span, node_right_span) = match node {
+            AstKind::BinaryExpression(expr) => {
+                (expr.operator.as_str(), expr.left.span(), expr.right.span())
+            }
+            AstKind::LogicalExpression(expr) => {
+                (expr.operator.as_str(), expr.left.span(), expr.right.span())
+            }
+            _ => unreachable!(),
+        };
+        // Since we don't store the exact span of the operators, approximate that span to be between the lhs
+        // and rhs of the expression.
+        let node_operator_span = Span::new(node_left_span.end + 1, node_right_span.start - 1);
+
+        let (parent_operator, parent_left_span, parent_right_span) = match parent {
+            AstKind::BinaryExpression(expr) => {
+                (expr.operator.as_str(), expr.left.span(), expr.right.span())
+            }
+            AstKind::LogicalExpression(expr) => {
+                (expr.operator.as_str(), expr.left.span(), expr.right.span())
+            }
+            AstKind::ConditionalExpression(expr) => {
+                // For conditional operators, the span covers both ? and :
+                ("?:", expr.test.span(), expr.alternate.span())
+            }
+            _ => unreachable!(),
+        };
+        let parent_operator_span = Span::new(parent_left_span.end + 1, parent_right_span.start - 1);
+
+        let (node_operator, node_precedence) = operator_and_precedence(node_operator).unwrap();
+        let (parent_operator, parent_precedence) =
+            operator_and_precedence(parent_operator).unwrap();
+        if self.in_the_same_group(node_operator, parent_operator)
+            && !(self.allow_same_precedence && node_precedence == parent_precedence)
+        {
+            // Report error at both operators
+            ctx.diagnostic(NoMixedOperatorsDiagnostic(
+                node_operator,
+                parent_operator,
+                node_operator_span,
+            ));
+            ctx.diagnostic(NoMixedOperatorsDiagnostic(
+                node_operator,
+                parent_operator,
+                parent_operator_span,
+            ));
+        }
```

Comment:
I think it would be more readable if we add another diagnostic with two spans, what do you think?

Reply:
It makes sense to me. I did this initially because eslint produces two errors at each operator respectively. Now I'm generating diagnostics like:
```
× eslint(no-mixed-operators): Disallow mixed binary operators
   ╭─[no_mixed_operators.tsx:1:1]
 1 │ a + b ?? c
   ·   ┬   ─┬
   ·   │    ╰──  with ??. Use parentheses to clarify the intended order of operations.
   ·   ╰── Unexpected mix of +
   ╰────
```
Does this look good to you?